### DISCUSSION TITLE
How to do equivalent of an irfft?

### DISCUSSION THREAD

**Discussion by OliverDudgeon** [JuliaMath/NFFT.jl#124] [Q&A] [answered]
Hi, thanks for your work on this package. I was wondering if it's possible, how I could do an `irfft`. For example, currently for an equidistantly spaced signal I do the following:

```julia
using FFTW
irfft(data, 2length(data)-1)
```

What would be the closest analogue with NFFT.jl? `data` will eventually have a non-equidistant spacing.

**Comment by tknopp** [comment, marked as answer]
The inverse NFFT does not exist analytically. But there is an approximation that is outlined here: 
 https://juliamath.github.io/NFFT.jl/dev/tools/#Sampling-Density
Programing-wise you would first calculate the density weights, then apply them and finally do the adjoint NFFT.

We do this in MRIReco:
https://github.com/MagneticResonanceImaging/MRIReco.jl/blob/master/src/Reconstruction/DirectReconstruction.jl#L35
and the line afterwards.
